src/python/pants/backend/experimental/python/lockfile_metadata.py

```diff
@@ -113,7 +113,7 @@ def calculate_invalidation_digest(requirements: Iterable[str]) -> str:
     """Returns an invalidation digest for the given requirements."""
     m = hashlib.sha256()
     inputs = {
-        "requirements": list(requirements),
+        "requirements": sorted(requirements),
```

Comment:
These should probably be sorted and unique: A requirements set that passes in an identical requirement through a requirements file _and_ through the `extra_requirements` flag should treat it as though the duplicated requirement is only there once.